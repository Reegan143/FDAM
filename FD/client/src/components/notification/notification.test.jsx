import { describe, it, expect } from 'vitest';
import { create } from 'react-test-renderer';
import NotificationsPopup from './notification';
import React from 'react';

const sampleNotifications = [
  {
    _id: '1',
    ticketNumber: '12345',
    complaintType: 'Network Issue',
    createdAt: '2025-03-04T12:00:00Z',
    messages: 'Your complaint has been received.',
    isRead: false,
  },
  {
    _id: '2',
    ticketNumber: '67890',
    complaintType: 'Billing Issue',
    createdAt: '2025-03-03T15:30:00Z',
    messages: 'Your issue is being processed.',
    isRead: true,
  }
];

describe('NotificationsPopup Component', () => {
  it('renders correctly when visible', () => {
    const tree = create(
      <NotificationsPopup 
        showNotifications={true} 
        notifications={sampleNotifications} 
        markAsRead={() => {}} 
        onClose={() => {}} 
        unreadCount={1} 
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when hidden', () => {
    const tree = create(
      <NotificationsPopup 
        showNotifications={false} 
        notifications={[]} 
        markAsRead={() => {}} 
        onClose={() => {}} 
        unreadCount={0} 
      />
    ).toJSON();
    expect(tree).toBe(null);
  });

  it('renders no notifications message when list is empty', () => {
    const tree = create(
      <NotificationsPopup 
        showNotifications={true} 
        notifications={[]} 
        markAsRead={() => {}} 
        onClose={() => {}} 
        unreadCount={0} 
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
