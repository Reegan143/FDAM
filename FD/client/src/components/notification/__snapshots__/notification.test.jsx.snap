// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotificationsPopup Component > renders correctly when visible 1`] = `null`;

exports[`NotificationsPopup Component > renders no notifications message when list is empty 1`] = `null`;
